packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart

```diff
@@ -215,4 +217,77 @@ class Breakpoint {
 
     return isWidthActive && isHeightActive && isRightPlatform;
   }
+
+  /// Returns the currently active [Breakpoint].
+  static Breakpoint activeBreakpointOf(BuildContext context) {
+    final SlotLayout? slotLayout =
+        context.findAncestorWidgetOfExactType<SlotLayout>();
+
+    if (slotLayout != null) {
+      Breakpoint? fallbackBreakpoint;
+
+      for (final MapEntry<Breakpoint, SlotLayoutConfig?> config
+          in slotLayout.config.entries) {
+        if (config.key.isActive(context)) {
+          if (config.key.platform != null) {
+            return config.key;
+          } else {
+            fallbackBreakpoint ??= config.key;
+          }
+        }
+      }
+      if (fallbackBreakpoint != null) {
+        return fallbackBreakpoint;
+      }
+    }
```

Comment:
Consider spinning this off into a separate method that maybe returns the breakpoint from `SlotLayout` if available:

```dart
static Breakpoint maybeGetBreakpointFromSlotLayout(BuildContext context) {
    final SlotLayout? slotLayout =
        context.findAncestorWidgetOfExactType<SlotLayout>();
    Breakpoint? fallbackBreakpoint;
    if (slotLayout != null) {
      for (final MapEntry<Breakpoint, SlotLayoutConfig?> config
          in slotLayout.config.entries) {
        if (config.key.isActive(context)) {
          if (config.key.platform != null) {
            return config.key;
          } else {
            fallbackBreakpoint ??= config.key;
          }
        }
      }
    }
    return fallbackBreakpoint;
}

```